jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/ArrayByteBufferPool.java

```diff
@@ -508,6 +476,24 @@ private Pool<RetainableByteBuffer> getPool()
             return _pool;
         }
 
+        private void evict(long excess)
+        {
+            List<Pool.Entry<RetainableByteBuffer>> idle = _pool.stream().filter(Pool.Entry::isIdle).toList();
```

Comment:
`toList()` is expensive, `forEach()` or `iterator()` should be used instead.

Reply:
Oh, and with the current `CompoundPool.stream()` implementation, we favor evicting from the `ConcurrentPool` before the `QueuedPool`, while I think we should do the opposite.

Reply:
The problem with the stream is that I need a capturing lambda. I'll add an iterator to the pool instead

Reply:
Don't forget to prefer the `QueuedPool` over the `ConcurrentPool`.